scripts/heratape_backup.py

```diff
@@ -0,0 +1,330 @@
+
+    logger.info('Moving on from data selection to tape writing!')
+    #example entry in "joined_results" which is the main data object that will be used to create db records
+    # and do the actual TAR
+    '''
+    ({'store_name': 'herastore01-8',
+      'store_ssh_host': 'heralib.aoc.nrao.edu',
+      'parent_dirs': '2459977',
+      'name': 'zen.2459977.21630.sum.autos.uvh5',
+      'deletion_policy': 'disallowed',
+      'full_path_on_store': '/export/hera/herastore01-8/2459977/zen.2459977.21630.sum.autos.uvh5'},
+     {'name': 'zen.2459977.21630.sum.autos.uvh5',
+      'type': 'uvh5',
+      'create_time': 1675303951,
+      'obsid': 1359306701,
+      'size': 19859900,
+      'md5': 'e910dffca62c8c859aacd873d2c14c6b'},
+     {'obsid': 1359306701,
+      'start_time_jd': 2459977.21624788,
+      'stop_time_jd': 2459977.21678797,
+      'start_lst_hr': 3.39516861968927,
+      'session_id': 1359303644})
+    '''
+    #turn our list of joined results into filenames, obsids, jds, and sizes
+    files = []
+    sizes = []
+    obsids = []
+    start_jds = []
+    for F in joined_results:
+        files.append(F[0]['full_path_on_store'])
+        sizes.append(F[1]['size'])
+        obsids.append(F[2]['obsid'])
+        start_jds.append(F[2]['start_time_jd'])
+    logger.info(f'Backing up JDS {"".join([str(jd) for jd in jds_to_backup])}. Nfiles = {len(files)}, Total Size = {np.round(np.sum(sizes)/1e12,1)}TB')
+    def update_tape_usage(tapes,testing=False):
+        #input: dict  from query_tape_usage and a session for heratape db
+        #output: input dict with added key: usage (sum in bytes known to heratape for each tape)
+        for i in range(len(tapes)):
+            tapeusage = query_tape_usage(tapes[i]['tape_id'],testing=testing)
+            tapes[i]['usage'] =  tapeusage
+        return tapes
+    def select_empty_tape(drive_id,testing=False):
+        # Impliments tape selection logic
+        # drive 1 uses tape slots 1-12, drive 2 uses 13-24
+        # there needs to be one empty slot for shuffling and another is taken up by a cleaner tape
+        # ideally these are split between the two halves of the jukebox
+        # input: drive number 0 or 1
+        # get the next tape in the relevant half which is not full
+        # full being defined as > 50%
+        # return: slot, tape_id, and tape_usage (in bytes) to be used
+        TAPECAPACITY = 18e12 #TODO: replace this with the number in heratape.Tapes
+        assert(drive_id  in [0,1])
+        tape_archive = query_tape_jukebox() #get the list of tapes currently in the jukebox
+        tape_archive = update_tape_usage(tape_archive,testing=testing) #find out how full they are.
+        #throw out any that are too full
+        empty_tapes = []
+        for tape in tape_archive:
+            if tape['tape_id'].startswith('CLN'):continue#ignore the cleaner!
+            logger.info(tape['usage'])
+            if tape['usage'] < TAPECAPACITY *0.5:
```

Comment:
I'm confused about this. An empty tape should have usage that is **way** less than half the tape capacity, right?